src/main/java/mate/academy/hibernate/relations/service/impl/ActorServiceImpl.java

```diff
@@ -1,16 +1,24 @@
 package mate.academy.hibernate.relations.service.impl;
 
+import mate.academy.hibernate.relations.dao.ActorDao;
 import mate.academy.hibernate.relations.model.Actor;
 import mate.academy.hibernate.relations.service.ActorService;
 
 public class ActorServiceImpl implements ActorService {
+    private final ActorDao actorDao;
+
+    public ActorServiceImpl(ActorDao actorDao) {
+        this.actorDao = actorDao;
+    }
+
     @Override
     public Actor add(Actor actor) {
-        return null;
+        return actorDao.add(actor);
     }
 
     @Override
     public Actor get(Long id) {
-        return null;
+        return actorDao.get(id).orElseThrow(() ->
+                new RuntimeException("Actor not found with id: " + id));
```

Comment:
```suggestion
                new DataProcessingException("Actor not found with id: " + id));
```

same for other places

Reply:
Artur Sarakhman
[@Vadym Pantielieienko](https://github.com/mate-academy/hibernate-relations-hw/pull/931#participant:523592) привіт, давай почнемо з того, що видалимо sessionFactory з try with resources :) Також, на рівні сервісів потрібно генерувати іншу помилку, а не DataProcessingException, розберись з цим

previous task I got this comment. in that case, what is the difference between that task and this and how to understand when i should put this exception in services and when not
also here is a part of the task that said
- create your custom unchecked DataProcessingException and throw it in the catch block on the DAO layer

Reply:
ill leave with out changes for now
 let me know if I still need to fix it

Reply:
Leave as it is.
Throwing RuntimeException is bad practice, it's better to use custom one. For example EntityNotFoundException